### ISSUE TITLE
Invalid c3d when opening using Mokka and MotionBuilder

### ISSUE THREAD

**Issue by pariterre** [pyomeca/ezc3d#64]
I am moving a discussion that belongs to here : 
Dear ezc3d creators,

I am writing this message because I have trouble making even a single valid C3d file. My project is in C++ and includes only 3D point data and no analog data. 
I use MotionBuilder, MlsViewer and Mokka to check my files. MotionBuilder is able to load my c3d files, but contains differences with what I see when I print the files with ezc3d (they seem fine there). First, it displays the points in a different order than I specify in the PARAMETER section. Also, the points in the first few frames are put into somewhat random positions close to the origin, something that I have not seen in any of the example files from www.c3d.org. In MlsViewer, every residual value seems like the ones from the example invalid files (from www.c3d.org). Lastly, Mokka cannot even load my files. Even for an empty file, Mokka issues:

```
Bad data first block.
 Unknown analog resolution. Set by default to 12.
 Index out of range. Default value returned.
```
I have tried setting different values for the PARAMETER section, setting values for the residual of each point, or adding ANALOG channels to the file. Nothing seems to work..
So, I wanted to ask you if you have any suggestions for what's going on or what am I doing wrong?

Thank you in advance,

Kiriakos
________________________

Dear Kiriakos, 
May I suggest that you open an issue on GitHub, so future user will benefit from our discussion :) 
(You could copy paste the description of the problem)
I am not sure that I understand your problem. Do you have any code that you could share with me? Do you use ezc3d to create a c3d, modify a c3d or just loading it?
_______________________

Dear @pariterre ,
Thank you for your quick response. I use ezc3d to create a C3d file from scratch. Even empty c3d files, i.e., 
```C++
ezc3d::c3d c3d_empty;
c3d_empty.write("myPath/emptyC3d.c3d");
```
are invalid.. Any code I write, is like the example contained in ezc3d-dev\example\main.cpp. For example, something like the following won't work:

```C++
ezc3d::c3d c3d;

ezc3d::ParametersNS::GroupNS::Parameter pointRate("RATE");                                        pointRate.set(std::vector<float>() = { 15}, { 1 });
c3d.parameter("POINT", pointRate);

-------- other POINT parameters definition here --------

std::string label;                                                                                                                                                    
for (int i = 0; i < N_POINTS; ++i) {        
        label = std::string("P") + std::to_string(i);
        c3d.point(label);
    }

for (int i = 0; i < N_FRAMES; ++i) {
        ezc3d::DataNS::Frame f;
        ezc3d::DataNS::Points3dNS::Points pts;
        for (int j = 0; j < N_POINTS; ++j) {
            ezc3d::DataNS::Points3dNS::Point pt;
            std::string label;
            label = std::string("P") + allData[i].keypoints[j].id;
            pt.name(label);
            float x = allData[i].keypoints[j].X;
            float y = allData[i].keypoints[j].Y;
            float z = allData[i].keypoints[j].Z;
            //float conf = allData[i].keypoints[j].conf;
            pt.x(x);
            pt.y(y);
            pt.z(z);
           //pt.residual(conf);
            pts.point(pt);
        }
        f.add(pts);
        c3d.frame(f);
}
c3d.write(myPath/myC3d.c3d");
```
Best,
Kiriakos

**Comment by pariterre** [comment]
Ok!
That is surprising..
I will have a try myself and let you know quick!

**Comment by pariterre** [comment]
Issue found and solved by the PR #65
